torch/nn/modules/pooling.py

```diff
@@ -1084,6 +1084,10 @@ class AdaptiveAvgPool1d(_AdaptiveAvgPoolNd):
     Args:
         output_size: the target output size H
 
+    Shape:
+        - Input: :math:`(*, C, L_{in})`, where :math:`*` is 1 dimension or none.
+        - Output: :math:`(*, C, \text{output\_size})`.
```

Comment:
The docs above reference `H` as the output size, which is confusing. To match the other 1D pooling modules, I suggest using `L_{in}` and `L_{out}`, where `L_{out} = \text{output\_size}`

Reply:
Also wdyt of using `or` instead of `*` here, e.g. `Input: (N, C, L_{in}) or (C, L_{in})`?

Reply:
Would the output shape also list out the options?

```
Input: (N, C, L_{in}) or (C, L_{in})
Output: (N, C, L_{out}) or (C, L_{out})
```

I think that looks okay. For the future, if there are only two shape options, list them out?

Reply:
Yep exactly- there's some precedent for documenting multiple shapes like this (see `nn.Upsample` docs) and I think it's nice and readable

Reply:
Assuming we work out how to deal with `Upsample` + BC breaking, would we list out 3 shapes for no batch and 3 shapes for batches?

Reply:
Ouch, that's a good point- don't think we'd want to do that. For the sake of discussion, let me toss out an idea:
```
([N], C, L), where [N] is an optional batch dimension
```
wdyt about the readability of this? I think this case will be fairly common